src/main/scala/org/constellation/checkpoint/CheckpointBlockValidator.scala

```diff
@@ -160,6 +160,21 @@ class CheckpointBlockValidator[F[_]: Sync](
   ): F[ValidationResult[List[Transaction]]] =
     t.toList.traverse(validateTransaction).map(x => x.map(_.map(List(_)))).map(_.combineAll)
 
+  def validateTransactionOrdinalChain(t: Iterable[Transaction]): ValidationResult[List[Transaction]] = {
```

Comment:
There's already a logic that does this validation. I would suggest extracting it to external class and use it in both places. `CheckpointAcceptanceService#isCheckpointBlockAllowedForAcceptance`

Reply:
@tbekas Your validation checks if `hashes` form a chain via `lastTxRef` but here I check if `ordinals` are correct. I think we can use both validations anyway.

Reply:
We talked with @kpudlik offline, but just for the record, the existing validation in `CheckpointAcceptanceService#isCheckpointBlockAllowedForAcceptance` checks both `prevHash` and `ordinal` of any transaction that's in the chain.

Reply:
Yep, I can confirm that @tbekas is right. I have extracted @tbekas's logic and reused in all the places where it was duplicated.